GameDev/EntityFactory.cpp

```diff
@@ -3,27 +3,66 @@
 
 EntityFactory::EntityFactory(b2World& b2world) : world(b2world)
 {
-	registery =	std::unordered_map<EntityType, Entity>{
-			{ EntityType::ENTITY, Entity() },
-			{ EntityType::ACTOR, Actor() },
-			{ EntityType::NPC, Npc() },
-			{ EntityType::PLAYER, Player() }
+	entityRegistery =	std::unordered_map<EntityType, Entity*>{
+			{ EntityType::ENTITY,new Entity() },
+			{ EntityType::ACTOR, new Actor() },
+			{ EntityType::NPC, new Npc() },
+			{ EntityType::PLAYER, new Player() }
+	};
+	b2BodyDef entDef = b2BodyDef();
+	entDef.type = b2BodyType::b2_staticBody;
+	b2BodyDef ActorDef;
+	ActorDef.type = b2BodyType::b2_dynamicBody;
+	b2BodyDef NpcDef;
+	NpcDef.type = b2BodyType::b2_dynamicBody;
+	b2BodyDef PlayerDef;
+	PlayerDef.type = b2BodyType::b2_dynamicBody;
+
+	bodyRegistery = std::unordered_map<EntityType, b2BodyDef>{
+			{ EntityType::ENTITY,  entDef },
+			{ EntityType::ACTOR,  ActorDef },
+			{ EntityType::NPC,  NpcDef },
+			{ EntityType::PLAYER,  PlayerDef }
 	};
 
 }
 
 EntityFactory::~EntityFactory()
 {
+	for (auto it = entityRegistery.begin(); it != entityRegistery.end(); ++it){
+		delete it->second;
```

Comment:
hier wordt toch de Entity* verwijderd? Heb je dan niet het probleem dat je nu een EntityType overhoudt die nergens meer aan is gekoppeld? Moet niet nog entityRegistery.clear() ofzo worden aangeroepen?